lgc/disassembler/Disassembler.cpp

```diff
@@ -226,66 +288,54 @@ void ObjDisassembler::processSection(ELFSectionRef sectionRef) {
   bool outputting = !(sectFlags & ELF::SHF_EXECINSTR);
   for (;;) {
     // One iteration of disassembling the section.
-    // Sort the symbols. (Stable sort as there may be duplicate addresses.)
-    stable_sort(symbols);
 
     // If AMDGPU, create a symbolizer, giving it the symbols.
     MCSymbolizer *symbolizerPtr = nullptr;
     if (m_objFile->getArch() == Triple::amdgcn) {
       std::unique_ptr<MCRelocationInfo> relInfo(m_target->createMCRelocationInfo(m_tripleName, *m_context));
       if (relInfo) {
-        std::unique_ptr<MCSymbolizer> symbolizer(
-            m_target->createMCSymbolizer(m_tripleName, nullptr, nullptr, &symbols, m_context, std::move(relInfo)));
+        std::unique_ptr<MCSymbolizer> symbolizer(m_target->createMCSymbolizer(
+            m_tripleName, nullptr, nullptr, &symbols.getSymbolizerSymbols(), m_context, std::move(relInfo)));
         symbolizerPtr = &*symbolizer;
         m_instDisassembler->setSymbolizer(std::move(symbolizer));
       }
     }
 
     // Disassemble the section contents.
+    size_t prevNumSymbols = symbols.size();
+    symbols.sort();
     tryDisassembleSection(sectionRef, sectType, sectFlags, outputting, symbols, relocs);
     if (outputting)
       break; // Done final outputting pass.
 
-    // If we created a symbolizer, get the referenced addresses from it and synthesize labels, avoiding
-    // duplicates. If there were no new referenced addresses, then we can do the final output in the next
-    // pass (indicated by the setting of outputting).
-    outputting = true;
-    if (symbolizerPtr && !symbolizerPtr->getReferencedAddresses().empty()) {
-      outputting = false;
-      std::vector<uint64_t> referencedAddresses;
-      referencedAddresses.insert(referencedAddresses.begin(), symbolizerPtr->getReferencedAddresses().begin(),
-                                 symbolizerPtr->getReferencedAddresses().end());
-      llvm::sort(referencedAddresses.begin(), referencedAddresses.end());
-      uint64_t lastLabel = ~uint64_t(0);
-      for (uint64_t labelAddr : referencedAddresses) {
-        if (labelAddr == lastLabel)
-          continue;
-        lastLabel = labelAddr;
-        synthesizedLabels.push_back({});
-        synthesizedLabels.back().reset(new std::string((Twine("_L") + Twine::utohexstr(labelAddr)).str()));
-        symbols.push_back(SymbolInfoTy(labelAddr, *synthesizedLabels.back(), ELF::STT_NOTYPE));
-      }
+    if (symbolizerPtr) {
+      for (uint64_t offset : symbolizerPtr->getReferencedAddresses())
```

Comment:
I guess offset and address are synonymous here? Wouldn't it be better to stick to either offset or address in this whole PR to keep things consistent and unambiguous?

Reply:
Ack.